tests/k8st/create_kind_cluster.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash -e
+
+# test directory.
+TEST_DIR=./tests/k8st
+
+# kubectl binary.
+: ${kubectl:=./kubectl}
+
+# kind binary.
+: ${KIND:=$TEST_DIR/kind}
+
+# Set config variables needed for ${kubectl}.
+export KUBECONFIG=~/.kube/kind-config-kind
+
+function checkModule(){
+  MODULE="$1"
+  echo "Checking kernel module $MODULE ..."
+  if lsmod | grep "$MODULE" &> /dev/null ; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+function load_image() {
+    local node=$1
+    docker cp ./calico-node.tar ${node}:/calico-node.tar
+    docker exec -t ${node} ctr -n=k8s.io images import /calico-node.tar
+    docker exec -t ${node} rm /calico-node.tar
+}
+
+function enable_dual_stack() {
+  # Based on instructions in http://docs.projectcalico.org/master/networking/dual-stack.md
+  local yaml=$1
+  # add assign_ipv4 and assign_ipv6 to CNI config
+  sed -i -e '/"type": "calico-ipam"/r /dev/stdin' "${yaml}" <<EOF
+              "assign_ipv4": "true",
+              "assign_ipv6": "true"
+EOF
+  sed -i -e 's/"type": "calico-ipam"/"type": "calico-ipam",/' "${yaml}"
+
+  # And add all the IPV6 env vars
+  sed -i '/# Enable IPIP/r /dev/stdin' "${yaml}" << EOF
+            - name: IP6
+              value: "autodetect"
+            - name: CALICO_IPV6POOL_CIDR
+              value: "fd00:10:244::/64"
+            - name: CALICO_IPV6POOL_IPIP
+              value: "Never"
+            - name: CALICO_IPV6POOL_NAT_OUTGOING
+              value: "false"
+EOF
+  # update FELIX_IPV6SUPPORT=true
+  sed -i '/FELIX_IPV6SUPPORT/!b;n;c\              value: "true"' "${yaml}"
+}
+
+echo "kubernetes dualstack requires ipvs mode kube-proxy for the moment."
+MODULES=("ip_vs" "ip_vs_rr" "ip_vs_wrr" "ip_vs_sh" "nf_conntrack_ipv4")
+for m in "${MODULES[@]}"; do
+  checkModule $m
+  if [[ $? -eq 1 ]]; then
+    echo "Could not find kernel module $m. install it..."
+    sudo modprobe $m
+  fi
+done
+echo
+
+echo "Download kind executable with dual stack support"
+# We need to replace kind executable and node image 
+# with official release once dual stack is fully supported by upstream.
+curl -L https://github.com/song-jiang/kind/releases/download/dualstack-1.17.0/kind -o ${KIND}
+chmod +x ${KIND}
+exit 0
+
+echo "Create kind cluster"
+${KIND} create cluster --image songtjiang/kindnode-dualstack:1.17.0 --config - <<EOF
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+  disableDefaultCNI: true
+  podSubnet: "192.168.0.0/16,fd00:10:244::/64"
+  ipFamily: DualStack
+nodes:
+# the control plane node
+- role: control-plane
+- role: worker
+- role: worker
+- role: worker
```

Comment:
How about just 3 nodes (i.e. control_plane + 2 workers), so that the setup is more similar to what we had with DIND?  Then there will be fewer changes in the BIRD config below.

Reply:
I feel we could keep another worker node here. By doing this, we keep the same configuration as our CI setup. Future test cases may requires 3 worker nodes but more than that is unlikely. There is only 3 lines code added to BIRD config to support it.

Reply:
OK, fine.